### PR TITLE
feat(ui): add notifications to machine storage tab

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Added notifications to the storage tab based on the angular version. The only thing I've changed is showing the commissioning error by itself instead of showing every possible error notification, which felt like overkill.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to a few legacy machine storage tabs and check that the notifications match the new react version, except for machines that have not yet been commissioned in which case where there should only be one notification.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2155

## Screenshot
Screenshot_2020-11-10 illegal-finch maas storage bolla MAAS(1)

### CODE REVIEW

File: ui/src/app/machines/views/MachineDetails/MachineStorage/MachineStorage.tsx [hunk 1]
```diff
@@ -21,6 +22,7 @@ const MachineStorage = (): JSX.Element => {
   if (machine && "disks" in machine && "special_filesystems" in machine) {
     return (
       <>
+        <StorageNotifications id={id} />
```
Comment (huwshimi) [inline ui/src/app/machines/views/MachineDetails/MachineStorage/MachineStorage.tsx:25]: I put the summary notifications inside MachineDetails. I was thinking if they're at that level they'll be in a consistent spot and won't accidentally get wrapped in another element, what do you think?

https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/machines/views/MachineDetails/MachineDetails.tsx#L57

File: ui/src/app/store/machine/utils.ts [hunk 2]
```diff
@@ -69,6 +70,33 @@ export const useCanEdit = (
   );
 };
 
+/**
+ * Check whether a machine's status allows storage configuration.
+ * @param machine - A machine object.
+ * @returns Whether the machine's status allows storage configuration.
+ */
+export const useMachineStorageConfigurable = (
```
Comment (huwshimi) [inline ui/src/app/store/machine/utils.ts:78]: I don't think these need to be hooks as they don't do anything that a normal function can't.

File: ui/src/scss/_patterns_notifications.scss [hunk 3]
```diff
@@ -50,4 +50,9 @@
       margin-left: $spv-outer--small;
     }
   }
+
+  .p-notification + .p-notification,
+  [class^="p-notification--"] + [class^="p-notification--"] {
```
Comment (huwshimi) [inline ui/src/scss/_patterns_notifications.scss:55]: Is this something that should go back to Vanilla?

**Review by huwshimi — APPROVED** [review]
This looks good, a few comments.

File: ui/src/app/machines/views/MachineDetails/MachineStorage/MachineStorage.tsx (hunk 1 above)
Comment (Caleb-Ellis) [reply ui/src/app/machines/views/MachineDetails/MachineStorage/MachineStorage.tsx:25]: Yeah that sounds good!

File: ui/src/app/store/machine/utils.ts (hunk 2 above)
Comment (Caleb-Ellis) [reply ui/src/app/store/machine/utils.ts:78]: Ah you're right. Originally I had them extending `useCanEdit` but it turns out I needed them to be separate.

File: ui/src/scss/_patterns_notifications.scss (hunk 3 above)
Comment (Caleb-Ellis) [reply ui/src/scss/_patterns_notifications.scss:55]: I'm not sure tbh. It's not really a bug that's being overwritten, just... the default Vanilla spacing seems a bit too large for MAAS.

**Pull request merged**